homeassistant/components/sensibo/climate.py

```diff
@@ -102,244 +93,251 @@ async def async_setup_entry(
+            else HVAC_MODE_OFF
         )
-        self._attr_max_temp = (
-            self._temperatures_list[-1] if self._temperatures_list else super().max_temp
+
+    @property
+    def hvac_modes(self) -> list[str]:
+        """Return the list of available hvac operation modes."""
+        return [
+            SENSIBO_TO_HA[mode]
+            for mode in self.coordinator.data[self.unique_id]["hvac_modes"]
+        ]
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return self.coordinator.data[self.unique_id]["temp"]
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature we try to reach."""
+        return self.coordinator.data[self.unique_id]["target_temp"]
+
+    @property
+    def target_temperature_step(self) -> float | None:
+        """Return the supported step of target temperature."""
+        return self.coordinator.data[self.unique_id]["temp_step"]
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the fan setting."""
+        return self.coordinator.data[self.unique_id]["fan_mode"]
+
+    @property
+    def fan_modes(self) -> list[str] | None:
+        """Return the list of available fan modes."""
+        return self.coordinator.data[self.unique_id]["fan_modes"]
+
+    @property
+    def swing_mode(self) -> str | None:
+        """Return the swing setting."""
+        return self.coordinator.data[self.unique_id]["swing_mode"]
+
+    @property
+    def swing_modes(self) -> list[str] | None:
+        """Return the list of available swing modes."""
+        return self.coordinator.data[self.unique_id]["swing_modes"]
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return convert_temperature(
+            self.coordinator.data[self.unique_id]["temp_list"][0],
+            TEMP_CELSIUS,
+            self.temperature_unit,
         )
-        self._attr_temperature_unit = self._temperature_unit
 
-        self._attr_supported_features = 0
-        for key in self._ac_states:
-            if key in FIELD_TO_FLAG:
-                self._attr_supported_features |= FIELD_TO_FLAG[key]
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return convert_temperature(
+            self.coordinator.data[self.unique_id]["temp_list"][-1],
+            TEMP_CELSIUS,
+            self.temperature_unit,
+        )
 
-        self._attr_state = self._external_state or super().state
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.coordinator.data[self.unique_id]["available"] and super().available
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Add additional state attributes."""
+        return {
+            "calibration_temperature": self.coordinator.data[self.unique_id][
```

Comment:
What is the calibration temperature?

Reply:
You can manually per device offset the reported temperature and humidity.
As example if the device reports 22 but in reality it's 23 you can set +1 here and then the device will report closer to reality. I thought it would be worth showing them as extra attributes if someone has them set.

Reply:
This can be nicely implemented using `Number` entity with a slider that will allow controlling the offset and also show the current state

Reply:
Smart suggestion. Did not consider it to be changeable but probably it is. I remove it from the attributes and try to get it into numbers in upcoming PR instead.